### PR TITLE
Feature trend graph

### PR DESCRIPTION
**Pull request by DeepSJain**
Grade Trend Graph

- [x] Use Cases
- [x] Controller
- [x] View
- [x] Tests

### CODE REVIEW

**Review by AlanC12138 — APPROVED** [review]
seems good to me.

File: src/main/java/weightScheme/OrderedWeight.java [hunk 1]
```diff
@@ -71,4 +71,9 @@ public double getTotalWeight() {
                 .mapToDouble(weight -> weight.getTotalWeight())
                 .sum();
     }
+
+    @Override
+    public double getWeightOfEachInstance() {
+        return orderedWeights[0].getWeightOfEachInstance();
```
Comment (MrSunshineVS) [inline src/main/java/weightScheme/OrderedWeight.java:77]: This function does not work and only returns the weight of the highest impact work.

File: src/main/java/weightScheme/SimpleWeight.java [hunk 2]
```diff
@@ -14,9 +14,16 @@ public SimpleWeight(Weight weight) {
     @Override
     public double computeWeighted(double[] marks) {
         double exactWeightEarned = DoubleStream.of(marks)
+                .filter(mark -> mark >= 0)
                 .map(mark -> mark * weight.getWeightOfEachInstance())
                 .sum();
-        return exactWeightEarned / (marks.length * weight.getWeightOfEachInstance());
```
Comment (MrSunshineVS) [inline src/main/java/weightScheme/SimpleWeight.java:19]: You have changed the way marks are computed, which breaks compatibility with the rest of the program (outline and assessment's mark computation does not. Remember that if you'd like to add specific functionality for your own feature, NEVER edit existing functionality that was not authored by you unless you've discussed it with the original author. A better alternative is to create your own functions and use those as extensions, which leaves an opportunity to merge functionality later.

**Review by MrSunshineVS — CHANGES_REQUESTED** [review]
I cannot approve a merge of this branch as it has changed too much of everyone else's code to be compatible. If you wish to try and resubmit your code, I would suggest rolling back every commit that was in a file that wasn't a part of your task, and implement whatever changes you need as extensions to the program. Also, if you do wish to edit pre-existing work like this, please discuss with the original authors first to figure out how to work around your needs.

File: src/main/java/weightScheme/SimpleWeight.java (hunk 2 above)
Comment (DeepSJain) [reply src/main/java/weightScheme/SimpleWeight.java:19]: That code is not mine. I pulled from [1-feature-course-with-outline](https://github.com/CSC207-2022F-UofT/course-project-justbob/tree/1-feature-course-with-outline). I am waiting for pull request #20 to be merged before pulling from there and creating another pull request. I will close this one.

**Pull request closed without merging**